src/App.tsx

```diff
@@ -34,62 +35,110 @@ export function getReorderedGoods(
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((goodA: string, goodB: string): number => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return goodA.localeCompare(goodB);
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
 
-      <ul>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverseGoods = () => {
+    this.setState((state) => ({ isReversed: !state.isReversed }));
+  };
+
+  resetChange = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const sortedGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info',
+              { 'is-light': sortType !== SortType.ALPABET })}
+            onClick={this.sortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+          <button
+            type="button"
+            className={classNames('button is-success',
+              { 'is-light': sortType !== SortType.LENGTH })}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+          <button
+            type="button"
+            className={classNames('button is-warning',
+              { 'is-light': !isReversed })}
+            onClick={this.reverseGoods}
+          >
+            Reverse
+          </button>
+
+          {(sortType !== SortType.NONE || isReversed)
+            ? (
```

Comment:
better use logical operator && instead of ternary operator
so if first condition is true it will return second one,
and if first is false it will return false which will be equal of null